tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp

```diff
@@ -0,0 +1,262 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_TRIANG_MATVEC_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_TRIANG_MATVEC_HPP_
+
+namespace KokkosKernelsSTD {
+
+//
+// Overwriting triangular matrix-vector product: y = Ax
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+	 class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class Triangle,
+	 class DiagonalStorage,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+         class Layout_y>
+requires (Layout_A::template mapping<std::experimental::extents<numRows_A, numCols_A>>::is_always_unique())
+void triangular_matrix_vector_product(kokkos_exec<ExeSpace> /*kexe*/,
+				      std::experimental::mdspan<
+					ElementType_A,
+					std::experimental::extents<numRows_A, numCols_A>,
+					Layout_A,
+					std::experimental::default_accessor<ElementType_A>
+				      > A,
+				      Triangle /* tr */,
+				      DiagonalStorage d,
+				      std::experimental::mdspan<ElementType_x,
+					std::experimental::extents<ext_x>,
+					Layout_x,
+					std::experimental::default_accessor<ElementType_x>
+				      > x,
+				      std::experimental::mdspan<ElementType_y,
+					std::experimental::extents<ext_y>,
+					Layout_y,
+					std::experimental::default_accessor<ElementType_y>
+				      > y)
+{
+  // constraints
+  static_assert(A.rank() == 2);
+  static_assert(x.rank() == 1);
+  static_assert(y.rank() == 1);
+
+  // preconditions
+  if ( A.extent(0) != A.extent(1) ){
+    throw std::runtime_error("KokkosBlas: triangular_matrix_vector_product: A.extent(0) != A.extent(1) ");
+  }
+  if ( A.extent(1) != x.extent(0) ){
+    throw std::runtime_error("KokkosBlas: triangular_matrix_vector_product: A.extent(1) != x.extent(0) ");
+  }
+  if ( A.extent(0) != y.extent(0) ){
+    throw std::runtime_error("KokkosBlas: triangular_matrix_vector_product: A.extent(0) != y.extent(0) ");
+  }
+
+  auto A_view = Impl::mdspan_to_view(A);
+  auto x_view = Impl::mdspan_to_view(x);
+  auto y_view = Impl::mdspan_to_view(y);
+
+  auto ex = ExeSpace();
+
+  constexpr bool implicitUnitDiag =
+    std::is_same_v<DiagonalStorage, std::experimental::linalg::implicit_unit_diagonal_t>;
+
+  if constexpr (std::is_same_v<Triangle, std::experimental::linalg::upper_triangle_t>)
+  {
+
+  // this print is detected in the tests
+#if defined LINALG_ENABLE_TESTS
```

Comment:
I intended for these to be on for debug only or more precisely only when running tests. 
They are buried in the kokkos impl so should not affect the base code. Also, I added them because they are detected by the tests to ensure that the proper dispatch to kokkos happens. 
It is somewhat easy to make a mistake such that one thinks the kokkos impl is called but actually the serial impl is the one being called. I was thinking that I can guard the iostream inclusion too.